src/NATS.Client.ObjectStore/NatsObjStore.cs

```diff
@@ -603,7 +603,7 @@ public async ValueTask DeleteAsync(string key, CancellationToken cancellationTok
 
     private async ValueTask PublishMeta(ObjectMetadata meta, CancellationToken cancellationToken)
     {
-        var ack = await JetStreamContext.PublishAsync(GetMetaSubject(meta.Name), meta, serializer: NatsObjJsonSerializer<ObjectMetadata>.Default, headers: NatsRollupHeaders, cancellationToken: cancellationToken);
+        var ack = await JetStreamContext.PublishAsync(GetMetaSubject(meta.Name), meta, serializer: NatsObjJsonSerializer<ObjectMetadata>.Default, headers: _natsRollupHeaders, cancellationToken: cancellationToken);
```

Comment:
maybe we should create the headers here in the local scope. I have a feeling if you call put twice with telemetry turned on you'd still get the same exception even with this fix applied. Could you also write a test showing the issue?

Reply:
You are correct. Calling this method twice will result in same exception.
I'll make the changes.